flink-tests/src/test/java/org/apache/flink/test/checkpointing/EventTimeAllWindowCheckpointingITCase.java

```diff
@@ -88,20 +80,17 @@ public void testTumblingTimeWindow() {
 		final int numElementsPerKey = 3000;
 		final int windowSize = 100;
 		final int numKeys = 1;
-		FailingSource.reset();
 
 		try {
 			StreamExecutionEnvironment env = StreamExecutionEnvironment.getExecutionEnvironment();
 			env.setParallelism(PARALLELISM);
 			env.setStreamTimeCharacteristic(TimeCharacteristic.EventTime);
 			env.enableCheckpointing(100);
-			env.setRestartStrategy(RestartStrategies.fixedDelayRestart(3, 0));
+			env.setRestartStrategy(RestartStrategies.fixedDelayRestart(1, 0));
 			env.getConfig().disableSysoutLogging();
 
 			env
-					.addSource(new FailingSource(numKeys,
-							numElementsPerKey,
-							numElementsPerKey / 3))
+					.addSource(new FailingSource(new EventTimeWindowCheckpointingITCase.KeyedEventTimeGenerator(numKeys, windowSize), numElementsPerKey))
```

Comment:
Reusing the small helper funs from other tests to me always seems problematic when the names of that test changes or when the functions change. Duplicating them here will duplicate code but in the long run is probably better. WDYT?

Reply:
I would start duplicating them if there is divergence and they start to fail :-)